### PR TITLE
feat(typings): add ts definitions

### PR DESCRIPTION
**Pull request by foray1010**
**What**: Added TypeScript definitions

**Why**: to pass ts type checking

**How**: I added typings/index.d.ts, but seems the eslint config doesn't work well with typescript, I have to ignored this file in .eslintignore to pass precommit hook. I see same problem in https://github.com/testing-library/react-testing-library/blob/master/typings/index.d.ts btw

**Checklist**:

- [x] Documentation
- [ ] Tests N/A
- [x] Ready to be merged

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+typings/index.d.ts
```
Comment (kentcdodds) [inline .eslintignore]: There's [a default eslintignore with kcd-scripts](https://github.com/kentcdodds/kcd-scripts/blob/master/src/config/eslintignore) and once you add an `.eslintignore` file, then the default no longer applies. Please include the files from the default here as well.

File: package.json [hunk 2]
```diff
@@ -19,7 +20,8 @@
     "setup": "npm install && npm run validate -s"
   },
   "files": [
-    "dist"
+    "dist",
+    "typings"
```
Comment (kentcdodds) [inline package.json]: Could you rebase your branch on master? There will probably be a small conflict here.

File: typings/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+interface Options {
```
Comment (kentcdodds) [inline typings/index.d.ts]: Looks good to me!

Note that there actually is a return value in the source code (though it's not documented). It's a "restore" function which restores useEffect to its original definition.

**Comment by foray1010** [comment]
@kentcdodds updated!

**Comment by kentcdodds** [comment]
@all-contributors please add @foray1010 for code

**Comment by kentcdodds** [comment]
Thanks so much for your help! I've added you as a collaborator on the project. Please make sure that you review the `other/MAINTAINING.md` and `CONTRIBUTING.md` files (specifically the bit about the commit messages and the git hooks) and familiarize yourself with the code of conduct (we're using [the contributor covenant](http://contributor-covenant.org/)). You might also want to [watch the repo](https://help.github.com/articles/watching-repositories/) to be notified when someone files an issue/PR. Please continue to make PRs as you feel the need (you can make your branches directly on the repo rather than your fork if you want). Thanks! And welcome to the team :)

**Pull request merged**